samcli/lib/intrinsic_resolver/intrinsics_symbol_table.py

```diff
@@ -0,0 +1,403 @@
+"""
+The symbol table that is used in IntrinsicResolver in order to resolve runtime attributes
+"""
+import os
+import uuid
+from random import randint
+
+from six import string_types
+
+from samcli.commands.local.lib.sam_base_provider import SamBaseProvider
+from samcli.lib.intrinsic_resolver.intrinsic_property_resolver import IntrinsicResolver
+from samcli.lib.intrinsic_resolver.invalid_intrinsic_exception import InvalidSymbolException
+
+
+class IntrinsicsSymbolTable(object):
+    AWS_ACCOUNT_ID = "AWS::AccountId"
+    AWS_NOTIFICATION_ARN = "AWS::NotificationArn"
+    AWS_PARTITION = "AWS::Partition"
+    AWS_REGION = "AWS::Region"
+    AWS_STACK_ID = "AWS::StackId"
+    AWS_STACK_NAME = "AWS::StackName"
+    AWS_URL_PREFIX = "AWS::URLSuffix"
+    AWS_NOVALUE = "AWS::NoValue"
+    SUPPORTED_PSEUDO_TYPES = [
+        AWS_ACCOUNT_ID,
+        AWS_NOTIFICATION_ARN,
+        AWS_PARTITION,
+        AWS_REGION,
+        AWS_STACK_ID,
+        AWS_STACK_NAME,
+        AWS_URL_PREFIX,
+        AWS_NOVALUE
+    ]
+
+    DEFAULT_REGION = "us-east-1"
+    REGIONS = {"us-east-1": ["us-east-1a", "us-east-1b", "us-east-1c", "us-east-1d", "us-east-1e", "us-east-1f"],
+               "us-west-1": ["us-west-1b", "us-west-1c"],
+               "eu-north-1": ["eu-north-1a", "eu-north-1b", "eu-north-1c"],
+               "ap-northeast-3": ["ap-northeast-3a"],
+               "ap-northeast-2": ["ap-northeast-2a", "ap-northeast-2b", "ap-northeast-2c"],
+               "ap-northeast-1": ["ap-northeast-1a", "ap-northeast-1c", "ap-northeast-1d"],
+               "sa-east-1": ["sa-east-1a", "sa-east-1c"],
+               "ap-southeast-1": ["ap-southeast-1a", "ap-southeast-1b", "ap-southeast-1c"],
+               "ca-central-1": ["ca-central-1a", "ca-central-1b"],
+               "ap-southeast-2": ["ap-southeast-2a", "ap-southeast-2b", "ap-southeast-2c"],
+               "us-west-2": ["us-west-2a", "us-west-2b", "us-west-2c", "us-west-2d"],
+               "us-east-2": ["us-east-2a", "us-east-2b", "us-east-2c"],
+               "ap-south-1": ["ap-south-1a", "ap-south-1b", "ap-south-1c"],
+               "eu-central-1": ["eu-central-1a", "eu-central-1b", "eu-central-1c"],
+               "eu-west-1": ["eu-west-1a", "eu-west-1b", "eu-west-1c"],
+               "eu-west-2": ["eu-west-2a", "eu-west-2b", "eu-west-2c"],
+               "eu-west-3": ["eu-west-3a", "eu-west-3b", "eu-west-3c"],
+               "cn-north-1": [],
+               "us-gov-west-1": []}
+
+    DEFAULT_PARTITION = "aws"
```

Comment:
I wanted to avoid using boto3 for this, because it may bring in authentication constraints. There are ways to get around it, but it seems very ugly/hacky.